src/FileType_HDF5.f90

```diff
@@ -6571,5 +6595,338 @@ SUBROUTINE compute_chunk_size(gdims,cdims)
+       attr_val=buf
+    END SUBROUTINE read_attribute_st0
+#endif
+!
+!-------------------------------------------------------------------------------
+!> @brief Reads a integer value attribute from a known dataset
+!> 
+!> @param obj_name the relative path to the dataset
+!> @param attr_name the desired name of the attribute
+!> @param attr_value the desired value of the attrbute
+!>  
+    SUBROUTINE read_attribure_i0(this,obj_name,attr_name,attr_val)
+       CHARACTER(LEN=*),PARAMETER :: myName='read_attribute_i0_HDF5FileType'
+       CLASS(HDF5FileType),INTENT(INOUT) :: this
+       CHARACTER(LEN=*),INTENT(IN) :: obj_name, attr_name
+       INTEGER(SNK),INTENT(INOUT) :: attr_val
+        
+#ifdef FUTILITY_HAVE_HDF5
+       INTEGER(HID_T) :: attr_id, dset_id
+       INTEGER(HSIZE_T),DIMENSION(1) :: dims
+       dims(1)=1
+
+       !Prepare the File and object for the attribute
+       CALL open_object(this,obj_name,dset_id)
+       CALL pre_read_attribute(this,dset_id,attr_name,attr_id)
+
+       CALL h5aread_f(attr_id,H5T_NATIVE_INTEGER,attr_val,dims,error)
+       IF(error /= 0) THEN 
+         CALL this%e%raiseError(modName//'::'//myName// &
+          ' - Failed to read attribute.')
+         RETURN
+       ENDIF
+
+       CALL post_attribute(this,attr_id,dset_id)
+#endif
+    END SUBROUTINE read_attribure_i0
+!
+!-------------------------------------------------------------------------------
+!> @brief Reads a double value attribute from a known dataset
+!> 
+!> @param obj_name the relative path to the dataset
+!> @param attr_name the desired name of the attribute
+!> @param attr_value the desired value of the attrbute
+!>  
+    SUBROUTINE read_attribute_d0(this,obj_name,attr_name,attr_val)
+       CHARACTER(LEN=*),PARAMETER :: myName='read_attribute_d0_HDF5FileType'
+       CLASS(HDF5FileType),INTENT(INOUT) :: this
+       CHARACTER(LEN=*),INTENT(IN) :: obj_name, attr_name
+       REAL(SDK),INTENT(INOUT) :: attr_val
+        
+#ifdef FUTILITY_HAVE_HDF5
+       INTEGER(HID_T) :: attr_id, dset_id
+       INTEGER(HSIZE_T),DIMENSION(1) :: dims
+       dims(1)=1
+
+       !Prepare the File and object for the attribute
+       CALL open_object(this,obj_name,dset_id)
+       CALL pre_read_attribute(this,dset_id,attr_name,attr_id)
+
+       CALL h5aread_f(attr_id,H5T_NATIVE_DOUBLE,attr_val,dims,error)
+       IF(error /= 0) THEN 
+         CALL this%e%raiseError(modName//'::'//myName// &
+          ' - Failed to read attribute.')
+         RETURN
+       ENDIF
+
+       CALL post_attribute(this,attr_id,dset_id)
+#endif
+    END SUBROUTINE read_attribute_d0
+!
+!-------------------------------------------------------------------------------
+!> @brief Sets up all attribute operations by checking links and opening object` 
+!> 
+!> @param obj_name the relative path to the dataset
+!> @param dset_id the HDF5 system id for the working dataset 
+!>  
+#ifdef FUTILITY_HAVE_HDF5
+    SUBROUTINE open_object(this,obj_name,dset_id)
```

Comment:
Also super nit-picky. Change dset_id to obj_id throughout, where relevant, for consistency.